packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -228,6 +230,92 @@ export const saveDataset = async (
 	return createDatasetFromSimulationResult(projectId, simulationId, datasetName);
 };
 
+/**
+ * This is a client side function to create a CsvAsset similar to the getCsv function from DatasetResource.java
+ * The reason we reimplement this client side is because runResults already has all of the data we need, except
+ * it's not in the correct format necessary to be consumed by the TeraDatasetDatatable component.
+ * @param runResults
+ * @param runId
+ * @returns CsvAsset object with the data from the runResults
+ */
+const createCsvAssetFromRunResults = (runResults: RunResults, runId?: string): CsvAsset | null => {
+	const runResult: RunResults = _.cloneDeep(runResults);
+	const runIdList = runId ? [runId] : Object.keys(runResult);
+	if (runIdList.length === 0) return null;
+
+	const csvColHeaders = Object.keys(runResult[runIdList[0]][0]);
+	let csvData: CsvAsset = {
+		headers: csvColHeaders,
+		data: [],
+		csv: [csvColHeaders],
+		rowCount: 0,
+		stats: []
+	};
+
+	const csvColumns: { [key: string]: number[] } = {};
+
+	runIdList.forEach((id) => {
+		csvData = {
+			...csvData,
+			data: [...csvData.data, ...(runResult[id] as any)],
+			rowCount: csvData.rowCount + runResult[id].length
+		};
+		runResult[id].forEach((row) => {
+			const rowValues = Object.values(row);
+			csvData.csv.push(rowValues as any);
+
+			csvColHeaders.forEach((header, index) => {
+				if (!csvColumns[header]) {
+					csvColumns[header] = [];
+				}
+				csvColumns[header].push(+rowValues[index]);
+			});
+		});
+	});
+
+	csvColHeaders.forEach((header) => {
```

Comment:
Now that I've said that the call should be made up here one level higher to avoid having a rest call inside a tight loop - perhaps an endpoint to receive `csvColHeaders` which uses the existing helper?